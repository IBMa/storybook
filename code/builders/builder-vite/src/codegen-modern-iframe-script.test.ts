import { describe, it, expect } from 'vitest';
import { generateModernIframeScriptCodeFromPreviews } from './codegen-modern-iframe-script';

const projectRoot = 'projectRoot';

describe('generateModernIframeScriptCodeFromPreviews', () => {
  it('handle one annotation', async () => {
    const result = await generateModernIframeScriptCodeFromPreviews({
      previewAnnotations: ['previewAnnotations'],
      projectRoot,
      frameworkName: 'frameworkName',
    });
    expect(result).toMatchSnapshot();
  });
  it('handle multiple annotations', async () => {
    const result = await generateModernIframeScriptCodeFromPreviews({
      previewAnnotations: ['previewAnnotations1', 'previewAnnotations2'],
      projectRoot,
      frameworkName: 'frameworkName',
    });
    expect(result).toMatchSnapshot();
  });
});
