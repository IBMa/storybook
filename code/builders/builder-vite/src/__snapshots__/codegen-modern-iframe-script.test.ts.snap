// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateModernIframeScriptCodeFromPreviews > handle multiple annotations 1`] = `
"import { composeConfigs, PreviewWeb, ClientApi } from 'storybook/internal/preview-api';
import '/virtual:/@storybook/builder-vite/setup-addons.js';
import { importFn } from '/virtual:/@storybook/builder-vite/storybook-stories.js';
import previewAnnotations1_2032 from "/previewAnnotations1";
import previewAnnotations2_2033 from "/previewAnnotations2";
const getProjectAnnotations = (hmrPreviewAnnotationModules = []) => {
  const configs = [
    hmrPreviewAnnotationModules.at(0) ?? previewAnnotations1_2032,
    hmrPreviewAnnotationModules.at(1) ?? previewAnnotations2_2033
  ]
  return composeConfigs(configs);
}

window.__STORYBOOK_PREVIEW__ = window.__STORYBOOK_PREVIEW__ || new PreviewWeb(importFn, getProjectAnnotations);

window.__STORYBOOK_STORY_STORE__ = window.__STORYBOOK_STORY_STORE__ || window.__STORYBOOK_PREVIEW__.storyStore;

if (import.meta.hot) {
  import.meta.hot.accept('/virtual:/@storybook/builder-vite/storybook-stories.js', (newModule) => {
    // importFn has changed so we need to patch the new one in
    window.__STORYBOOK_PREVIEW__.onStoriesChanged({ importFn: newModule.importFn });
  });

  import.meta.hot.accept(["/previewAnnotations1","/previewAnnotations2"], (previewAnnotationModules) => {
    // getProjectAnnotations has changed so we need to patch the new one in
    window.__STORYBOOK_PREVIEW__.onGetProjectAnnotationsChanged({ getProjectAnnotations: () => getProjectAnnotations(previewAnnotationModules) });
  });
};"
`;

exports[`generateModernIframeScriptCodeFromPreviews > handle one annotation 1`] = `
"import { composeConfigs, PreviewWeb, ClientApi } from 'storybook/internal/preview-api';
import '/virtual:/@storybook/builder-vite/setup-addons.js';
import { importFn } from '/virtual:/@storybook/builder-vite/storybook-stories.js';
import previewAnnotations_1983 from "/previewAnnotations";
const getProjectAnnotations = (hmrPreviewAnnotationModules = []) => {
  const configs = [
    hmrPreviewAnnotationModules.at(0) ?? previewAnnotations_1983
  ]
  return composeConfigs(configs);
}

window.__STORYBOOK_PREVIEW__ = window.__STORYBOOK_PREVIEW__ || new PreviewWeb(importFn, getProjectAnnotations);

window.__STORYBOOK_STORY_STORE__ = window.__STORYBOOK_STORY_STORE__ || window.__STORYBOOK_PREVIEW__.storyStore;

if (import.meta.hot) {
  import.meta.hot.accept('/virtual:/@storybook/builder-vite/storybook-stories.js', (newModule) => {
    // importFn has changed so we need to patch the new one in
    window.__STORYBOOK_PREVIEW__.onStoriesChanged({ importFn: newModule.importFn });
  });

  import.meta.hot.accept(["/previewAnnotations"], (previewAnnotationModules) => {
    // getProjectAnnotations has changed so we need to patch the new one in
    window.__STORYBOOK_PREVIEW__.onGetProjectAnnotationsChanged({ getProjectAnnotations: () => getProjectAnnotations(previewAnnotationModules) });
  });
};"
`;
